keras_nlp/models/gpt_neo_x/gpt_neo_x_decoder.py

```diff
@@ -0,0 +1,219 @@
+        max_sequence_length=512,
+        name=None,
+        **kwargs,
+    ):
+        self._input_shape = kwargs.pop("build_input_shape", None)
+
+        super().__init__(name=name, **kwargs)
+        self.intermediate_dim = intermediate_dim
+        self.num_heads = num_heads
+        self.dropout = dropout
+        self.rotary_percentage = rotary_percentage
+        self.rotary_max_wavelength = rotary_max_wavelength
+        self.max_sequence_length = max_sequence_length
+        self.activation = keras.activations.get(activation)
+        self.layer_norm_epsilon = layer_norm_epsilon
+        self.kernel_initializer = keras.initializers.get(kernel_initializer)
+        self.bias_initializer = keras.initializers.get(bias_initializer)
+        self._built = False
+        self.supports_masking = True
+        self.rotary_percentage = rotary_percentage
+
+        if self._input_shape is not None:
+            self._build(self._input_shape)
+
+    def _build(self, input_shape):
+        # Create layers based on input shape.
+        self._built = True
+        self._input_shape = input_shape
+        # Infer the dimension of our hidden feature size from the build shape.
+        hidden_dim = input_shape[-1]
+
+        # Self attention layers.
+        self._self_attention_layer = GPTNeoXAttention(
+            num_heads=self.num_heads,
+            hidden_dim=hidden_dim,
+            dropout=self.dropout,
+            rotary_percentage=self.rotary_percentage,
+            rotary_max_wavelength=self.rotary_max_wavelength,
+            max_sequence_length=self.max_sequence_length,
+            kernel_initializer=clone_initializer(self.kernel_initializer),
+            bias_initializer=clone_initializer(self.bias_initializer),
+        )
+
+        self._self_attention_layernorm = keras.layers.LayerNormalization(
+            epsilon=self.layer_norm_epsilon,
+        )
+
+        self._self_attention_dropout = keras.layers.Dropout(
+            rate=self.dropout,
+        )
+
+        # Feedforward layers.
+        self._feedforward_intermediate_dense = keras.layers.Dense(
+            self.intermediate_dim,
+            activation=self.activation,
+            kernel_initializer=clone_initializer(self.kernel_initializer),
+            bias_initializer=clone_initializer(self.bias_initializer),
+        )
+        self._feedforward_output_dense = keras.layers.Dense(
+            hidden_dim,
+            kernel_initializer=clone_initializer(self.kernel_initializer),
+            bias_initializer=clone_initializer(self.bias_initializer),
+        )
+
+        self._feedforward_layernorm = keras.layers.LayerNormalization(
+            epsilon=self.layer_norm_epsilon,
+        )
+
+        self._feedforward_dropout = keras.layers.Dropout(
+            rate=self.dropout,
+        )
+
+    def call(
+        self,
+        decoder_sequence,
+        decoder_padding_mask=None,
+        decoder_attention_mask=None,
+    ):
+        if not self._built:
```

Comment:
Do things work if you just make this a normal `build` method and delete these lines? It should. We only need the weird indirection for TransformerDecoder because it supports multiple inputs. That's not the case here.

Delete this, and rename `_build` -> `build`

Reply:
Thanks for the clarification !
It works by using normal `build`